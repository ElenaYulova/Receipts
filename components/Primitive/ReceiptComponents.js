import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptComponents.css';

// Компонент Ингредиенты рецепта кулинарной книги. Переиспользуемый
export default class ReceiptComponents extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        components: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
        let text = this.props.components.split(".");
        text = text.map((part, i) => <li  key={i}>{part}</li> );
    return <div className="receipt-components">
        <ul  id = {name} >{text}</ul>      
    </div>
 }   
}