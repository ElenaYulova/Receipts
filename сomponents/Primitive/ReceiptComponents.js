import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptComponents.css';

// Компонент Ингредиенты рецепта кулинарной книги. Переиспользуемый
export default class ReceiptComponents extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        components: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
        const text = this.props.steps.split(".");
        text = text.map((part, i) => <li  key={i}>{part}</li> );
    return <div className="receipt-components">
        <ul  id = {name} >{text}</ul>      
    </div>
 }   
}