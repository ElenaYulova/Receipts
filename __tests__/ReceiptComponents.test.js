"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import ReceiptComponents from '../components/Primitive/ReceiptComponents';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('>>>ReceiptComponents --- Shallow Render REACT COMPONENTS',()=>{
    //shapshots
     
     
      const receiptComponent = <ReceiptComponents
      name = "name1"
      components = "sfADFASDFSA"
       />;
    
         
         const component = renderer.create(receiptComponent);
    
         it('renders receipt', () => {
           let componentTree = component.toJSON();
           expect(componentTree).toMatchSnapshot();
         });
    
         
    });