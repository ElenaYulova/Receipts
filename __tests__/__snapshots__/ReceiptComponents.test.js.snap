// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ReceiptComponents --- Shallow Render REACT COMPONENTS renders receipt 1`] = `
<div
  className="receipt-components"
>
  <ul
    id="name1"
  >
    <li>
      sfADFASDFSA
    </li>
  </ul>
</div>
`;
